src/Microsoft.ML.Data/Scorers/SchemaBindablePredictorWrapper.cs

```diff
@@ -426,9 +426,15 @@ private protected override bool SaveAsOnnxCore(OnnxContext ctx, RoleMappedSchema
 
             var mapper = ValueMapper as ISingleCanSaveOnnx;
             Contracts.CheckValue(mapper, nameof(mapper));
-            Contracts.Assert(schema.Feature.HasValue);
             Contracts.Assert(Utils.Size(outputNames) == 3); // Predicted Label, Score and Probablity.
 
+            // Prior doesn't have a feature column and uses the training label column to determine predicted labels
+            if (mapper.ToString().EndsWith("PriorModelParameters")) {
```

Comment:
I know I had asked you to explore the option of checking the mapper type here but I didn't mean check it as a string. I meant check the type directly (using typeof()). But now I don't think that is good idea either because we might introduce a bad dependency. 
I think the better approach is to explicitly handle two cases (if the feature column is present and otherwise).